test/unit-tests/reporter-handlers.test.js

```diff
@@ -0,0 +1,77 @@
+const sinon = require('sinon');
+const expect = require('chai').expect;
+const { EventEmitter } = require('events');
+
+// mock event emitter to simulate newman and socket-client
+const mockNewman = new EventEmitter();
+const mockSocket = new EventEmitter();
+
+const ioStub = sinon.stub().callsFake(() => {
+    return mockSocket;
+});
+
+// io() is a default export of socket.io-client; resolve it to export
+// a sinon stub instead
+require.cache[require.resolve('socket.io-client')] = {
+    exports: ioStub,
+};
```

Comment:
You can stub default export using [mock-require](https://www.npmjs.com/package/mock-require)?

Reply:
Thank you for this suggestion!